src/track/trackmetadatataglib.cpp

```diff
@@ -919,23 +1021,23 @@ bool writeTrackMetadataIntoID3v2Tag(TagLib::ID3v2::Tag* pTag,
         return false;
     }
 
-    writeTrackMetadataIntoTag(pTag, trackMetadata, false);
+    // NOTE(uklotzde): Setting the comment for ID3v2 tags does
+    // not work as expected when using TagLib 1.9.1 and must
+    // be skipped! Otherwise special purpose comment fields
+    // with a description like "iTunSMPB" might be overwritten.
+    // Mixxx implements a special case handling for ID3v2 comment
+    // frames (see below)
+    writeTrackMetadataIntoTag(pTag, trackMetadata,
+            WRITE_TAG_OMIT_TRACK_NUMBER | WRITE_TAG_OMIT_YEAR | WRITE_TAG_OMIT_COMMENT);
+
+    // Writing the common comments frame has been omitted (see above)
     writeID3v2CommentsFrame(pTag, trackMetadata.getComment());
 
-    // additional tags
-    writeID3v2TextIdentificationFrame(pTag, "TPE2",
-            trackMetadata.getAlbumArtist());
-    // According to the specification "The 'TBPM' frame contains the number
-    // of beats per minute in the mainpart of the audio. The BPM is an
-    // integer and represented as a numerical string."
-    // Reference: http://id3.org/id3v2.3.0
-    writeID3v2TextIdentificationFrame(pTag, "TBPM",
-            formatBpmInteger(trackMetadata), true);
-    writeID3v2TextIdentificationFrame(pTag, "TKEY", trackMetadata.getKey());
-    writeID3v2TextIdentificationFrame(pTag, "TCOM",
-            trackMetadata.getComposer());
-    writeID3v2TextIdentificationFrame(pTag, "TIT1",
-            trackMetadata.getGrouping());
+    writeID3v2TextIdentificationFrame(pTag, "TRCK",
+            TrackNumbers::joinStrings(
+                    trackMetadata.getTrackNumber(),
+                    trackMetadata.getTrackTotal()));
```

Comment:
Nevermind, it is guaranteed since we now always read the value from DB or track.